ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala

```diff
@@ -1073,15 +1074,18 @@ private class PostgresLedgerDao(
                     "package" -> p._1.toByteArray
                 )
               )
-            executeBatchSql(
-              SQL_INSERT_PACKAGE,
-              namedPackageParams
-            )
-            PersistenceResponse.Ok
-          }.recover {
-            case NonFatal(e) if e.getMessage.contains("duplicate key") =>
-              conn.rollback()
-              PersistenceResponse.Duplicate
+            import scala.math.max
+            val changeCount: Int = executeBatchSql(SQL_INSERT_PACKAGE, namedPackageParams)
+              .map(max(0, _))
```

Comment:
Why the `max(0, _)`? I can't find nice docs for `BatchSql`, but it looks like it should return the number of changed rows for each batch element (which should be exactly 0 or 1). Does it return something like -1 in some cases?

Reply:
Under the hood, anorm calls this method: https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#executeBatch()

It returns either a nonnegative number or one of two error codes, both of which are negative values. (One of those cases is `SUCCESS_NO_INFO`, so not sure what the right behavior is for this case… :-/ )